rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala

```diff
@@ -277,57 +289,56 @@ class CachedNodeChangesServiceImpl(
     ()
   }
 
-  //just delegate to the actual service
-  override def getCurrentValidIntervals(since: Option[DateTime]) = {
-    changeService.getCurrentValidIntervals(since)
-  }
 
-  override def getChangesByInterval(since: Option[DateTime]) = {
+  /*
+   * It's actually just using the cache
+   */
+  override def countChangesByRuleByInterval() = {
     for {
       changes <- initCache()
     } yield {
       changes
     }
   }
-}
-
-object NodeChanges {
 
-  // Filter changes on based on rules
-  def changesOnRule(ruleId : RuleId)(changes  : Map[Interval,Seq[ResultRepairedReport]]) = {
-    changes.mapValues(_.filter(_.ruleId == ruleId))
+  /**
+   * Get the changes for the given interval, without using the cache.
+   */
+  override def getChangesForInterval(ruleId: RuleId, interval: Interval, limit: Option[Int]): Box[Seq[ResultRepairedReport]] = {
+    for {
```

Comment:
well, certainly there was some other comands at some point of the dev, that were removed in the end (but not the for/yield  structure).